packages/live-share/src/EphemeralTimer.ts

```diff
@@ -3,239 +3,348 @@
  * Licensed under the Microsoft Live Share SDK License.
  */
 
-import {
-    DataObject,
-    DataObjectFactory,
-    DataObjectTypes,
-} from "@fluidframework/aqueduct";
-import { IEventThisPlaceHolder } from "@fluidframework/common-definitions";
-import { IEphemeralEvent, UserMeetingRole } from "./interfaces";
-import { EphemeralEventScope } from './EphemeralEventScope';
-import { EphemeralEventTarget } from './EphemeralEventTarget';
-import { TimeInterval } from './TimeInterval';
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { EphemeralEventScope } from "./EphemeralEventScope";
+import { EphemeralEventTarget } from "./EphemeralEventTarget";
+import { EphemeralObjectSynchronizer } from "./EphemeralObjectSynchronizer";
+import { IClientTimestamp, IEphemeralEvent, UserMeetingRole } from "./interfaces";
+import { IEvent } from "@fluidframework/common-definitions";
+import { cloneValue } from "./internals/utils";
 import { EphemeralEvent } from "./EphemeralEvent";
 
-/**
- * @hidden
- */
-export interface ITimerState {
-    timeStarted: number;
-    position: number;
-    duration: number;
-    running: boolean;
+/** for all time values millis from epoch is used */
+export interface ITimerConfig {
+  configChangedAt: number;
+  clientId: string;
+  duration: number;
+  // position when config change occured
+  position: number;
+  // running or not when config change occured
+  running: boolean;
 }
 
 /**
- * @hidden
+ * Events supported by `EphemeralTimer` object.
  */
- export interface IBeginTimerEvent extends IEphemeralEvent {
-    duration: number;
+ export enum EphemeralTimerEvents {
+  /**
+   * Timer has started
+   */
+  onStart = 'onStart',
```

Comment:
>onStart [](http://example.com/codeflow?start=2&length=7)

Sorry I'm just catching this... These names aren't consistent with our other events. Should be something like "timerStarted" to match other events like "presenceChanged"